python/ray/util/sgd/v2/tests/test_tune.py

```diff
@@ -127,6 +131,67 @@ def train_func(config):
         tune.run(TestTrainable)
 
 
+def test_tune_checkpoint(ray_start_2_cpus):
+    def train_func():
+        for i in range(10):
+            sgd.report(test=i)
+        sgd.save_checkpoint(hello="world")
```

Comment:
Seems like the three checkpointing tests follow different "patterns" (ordering and frequency) when calling `sgd.report` and `sgd.save_checkpoint`. Though functionally they may all work, do we want to conform to a recommended way of doing this, especially since `_current_iter` is used to define the iteration in the Tune trial?

Reply:
I don't think SGD should conform to a recommended way of doing this. The recommended way is the way that Tune recommends (these tests in fact are copied over from Tune tests), which could change in the future.

The only thing to document is there is `sgd.save_checkpoint()` is equivalent to `tune.checkpoint_dir` and `sgd.report()` is equivalent to `tune.report()`